src/examples/v5.tsx

```diff
@@ -0,0 +1,258 @@
+import {
+  ChakraProvider,
+  extendTheme,
+  Box,
+  VStack,
+  HStack,
+  Text,
+} from "@chakra-ui/react";
+import DepositButton from "../components/v1/DepositButton";
+import BridgeButton from "../components/v1/BridgeButton";
+import DepositAndBridgeButton from "../components/v1/DepositAndBridgeButton";
+import { createRoot } from "react-dom/client";
+import {
+  BoringVaultV1Provider,
+  useBoringVaultV1,
+} from "../contexts/v1/BoringVaultContextV1";
+import { WagmiProvider, createConfig, http } from "wagmi";
+import { mainnet } from "wagmi/chains";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import {
+  ConnectKitButton,
+  ConnectKitProvider,
+  getDefaultConfig,
+} from "connectkit";
+import { ethers } from "ethers";
+import { useEthersSigner } from "../hooks/ethers";
+
+const config = createConfig(
+  getDefaultConfig({
+    // Your dApps chains
+    chains: [mainnet],
+    transports: {
+      // RPC URL for each chain
+      [mainnet.id]: http(
+        `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`
+      ),
+    },
+
+    // Required API Keys
+    // ! https://cloud.walletconnect.com/sign-in
+    walletConnectProjectId: process.env.WALLETCONNECT_PROJECT_ID!,
+
+    // Required App Info
+    appName: "Boring Vault LayerZero Bridge Example",
+
+    // Optional App Info
+    appDescription: "Example app showcasing LayerZero bridging with Sonic USD Vault",
+    appUrl: "http://localhost:9000", // your app's url
+    appIcon: "https://family.co/logo.png", // your app's logo,no bigger than 1024x1024px (max. 1MB)
+  })
+);
+const ethersInfuraProvider = new ethers.InfuraProvider(
+  "mainnet",
+  process.env.INFURA_API_KEY
+);
+
+console.warn(process.env.WALLETCONNECT_PROJECT_ID);
+
+const queryClient = new QueryClient();
+
+// Customize the theme to fit your branding or design needs
+const theme = extendTheme({
+  colors: {
+    brand: {
+      100: "#f7fafc",
+      // ... (provide your brand colors)
+    },
+  },
+  components: {
+    Modal: {
+      baseStyle: (props: any) => ({
+        dialog: {
+          bg: "brand.100",
+        },
+      }),
+    },
+  },
+});
+
+const VaultWidget = () => {
+  const [assets, setAssets] = React.useState<number>(0);
+  const [userShares, setUserShares] = React.useState<number>(0);
+  const [shareValue, setShareValue] = React.useState<number>(0);
+  
+  const {
+    fetchTotalAssets,
+    isBoringV1ContextReady,
+    fetchUserShares,
+    fetchShareValue,
+    fetchUserUnlockTime,
+  } = useBoringVaultV1();
+  const signer = useEthersSigner();
+
+  useEffect(() => {
+    console.warn("ready: ", isBoringV1ContextReady);
+    if (!isBoringV1ContextReady || !signer) return;
+    
+    const fetchData = async () => {
+      try {
+        const [totalAssets, userSharesResult, shareValueResult] = await Promise.all([
+          fetchTotalAssets(),
+          fetchUserShares(await signer.getAddress()),
+          fetchShareValue()
+        ]);
```

Comment:
Let's call the signer outside the Promise.all for easier debugging, also there's not advantage within it.

```suggestion
        const signerAddress = await signer.getAddress();
        const [totalAssets, userSharesResult, shareValueResult] = await Promise.all([
          fetchTotalAssets(),
          fetchUserShares(signerAddress)
          fetchShareValue()
        ]);
```